Core/src/Batch/SysvConfigStorage.php

```diff
@@ -29,19 +29,46 @@ class SysvConfigStorage implements ConfigStorageInterface
 {
     const VAR_KEY = 1;
 
+    const DEFAULT_SHM_SIZE = 200000;
+
+    const DEFAULT_PERM = 0600;
+
+    const DEFAULT_PROJECT = 'A';
+
     /* @var int */
     private $sysvKey;
 
     /* @var int */
     private $semid;
 
+    /* @var int */
+    private $shmSize;
+
+    /* @var int */
+    private $perm;
+
+    /* @var string */
+    private $project;
+
     /**
      * Prepare the key for semaphore and shared memory.
      */
     public function __construct()
     {
-        $this->sysvKey = ftok(__FILE__, 'A');
-        $this->semid = sem_get($this->sysvKey, 1, 0600, 1);
+        $this->shmSize = intval(getenv('GOOGLE_CLOUD_BATCH_SHM_SIZE'));
+        if ($this->shmSize === 0) {
+            $this->shmSize = self::DEFAULT_SHM_SIZE;
+        }
+        $this->perm = octdec(getenv('GOOGLE_CLOUD_BATCH_PERM'));
```

Comment:
Not that anyone would really want to, but `octdec("0000") === 0` so the value would be set to `0600`

Reply:
That's ok